docs/pnpm.md

```diff
@@ -0,0 +1,285 @@
+
+To update the `pnpm-lock.yaml` file manually, either
+
+-   [install pnpm](https://pnpm.io/installation) and run `pnpm install --lockfile-only` or `pnpm import`
+-   use the Bazel-managed pnpm by running `bazel run -- @pnpm//:pnpm --dir $PWD install --lockfile-only` or `bazel run -- @pnpm//:pnpm --dir $PWD import`
+
+If the `ASPECT_RULES_JS_FROZEN_PNPM_LOCK` environment variable is set and `update_pnpm_lock` is True,
+the build will fail if the pnpm lock file needs updating.
+It is recommended to set this environment variable on CI when `update_pnpm_lock` is True.
+
+## Working with packages
+
+### Patching
+
+You can apply patches to packages you fetch remotely such as from npm.
+Use the `patches` and `patch_args` attributes of `npm_translate_lock`.
+These are designed to be similar to the same-named attributes of
+[http_archive](https://bazel.build/rules/lib/repo/http#http_archive-patch_args).
+
+Paths in patch files must be relative to the root of the package.
+If the version is left out of the package name, the patch will be applied to every version of the npm package.
+
+`patch_args` defaults to `-p0`, but `-p1` will usually be needed for patches generated by git.
+
+In case multiple entries in `patches` match, the list of patches are additive.
+(More specific matches are appended to previous matches.)
+However if multiple entries in `patch_args` match, then the more specific name matches take precedence.
+
+For example,
+
+```starlark
+npm_translate_lock (
+    ...
+    patches = {
+        "@foo/bar": ["//:patches/foo+bar.patch"],
+        "fum@0.0.1": ["//:patches/fum@0.0.1.patch"],
+    },
+    patch_args = {
+        "*": ["-p1"]
+        "@foo/bar": ["-p0"]
+        "fum@0.0.1": ["-p2"]
+    },
+)
+```
+
+### Lifecycles
+
+npm packages have "lifecycle scripts" such as `postinstall` which are documented here:
+<https://docs.npmjs.com/cli/v9/using-npm/scripts#life-cycle-scripts>
+
+We refer to these as "lifecycle hooks".
+
+> You can disable this feature completely by setting all packages to have no hooks, using
+> `lifecycle_hooks = { "*": [] }` in `npm_translate_lock`.
+
+Because rules_js models the execution of these hooks as build actions, rather than repository rules,
+the result can be stored in the remote cache and shared between developers.
+Typically these actions are not run in Bazel's action sandbox because of the overhead of setting up and tearing down the sandboxes.
+
+> You can set `lifecycle_hooks_no_sandbox = False` in `npm_translate_lock` to change this default.
+
+In addition to sandboxing, Bazel supports other `execution_requirements` for actions,
+in the attribute of <https://bazel.build/rules/lib/actions#run>.
+You can have control over these using the `lifecycle_hooks_execution_requirements` attribute of `npm_translate_lock`.
+
+Some hooks may fail to run under rules_js, and you don't care to run them.
+You can use the `lifecycle_hooks_exclude` attribute of `npm_translate_lock` to turn them off for a package,
```

Comment:
Could recommend using `lifecycle_hooks` here as well to disable for an individual package:
```
`lifecycle_hooks = { "fum@0.0.1": [] }` in `npm_translate_lock`
```

Reply:
"which is equivalent to setting the `lifecycle_hooks` to an empty list for that package." was the next line, I think that covers it